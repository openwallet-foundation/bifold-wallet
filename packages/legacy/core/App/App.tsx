import AgentProvider from '@credo-ts/react-hooks'
import * as React from 'react'
import { useEffect, useMemo } from 'react'
import { StatusBar } from 'react-native'
import SplashScreen from 'react-native-splash-screen'
import Toast from 'react-native-toast-message'

import { animatedComponents } from './animated-components'
import ErrorModal from './components/modals/ErrorModal'
import NetInfo from './components/network/NetInfo'
import toastConfig from './components/toast/ToastConfig'
import { Container, ContainerProvider } from './container-api'
import { ActivityProvider } from './contexts/activity'
import { AnimatedComponentsProvider } from './contexts/animated-components'
import { AuthProvider } from './contexts/auth'
import { NetworkProvider } from './contexts/network'
import { StoreProvider } from './contexts/store'
import { ThemeProvider } from './contexts/theme'
import { TourProvider } from './contexts/tour/tour-provider'
import { initLanguages, initStoredLanguage, translationResources } from './localization'
import RootStack from './navigators/RootStack'
import { theme } from './theme'
import { OpenIDCredentialRecordProvider } from './modules/openid/context/OpenIDCredentialRecordProvider'
import { tours } from './constants'

const App = (system: Container): React.FC => {
  initLanguages(translationResources)

  const AppComponent = () => {
    useMemo(() => {
      initStoredLanguage().then()
    }, [])

    useEffect(() => {
      // Hide the native splash / loading screen so that our
      // RN version can be displayed.
      SplashScreen.hide()
    }, [])

    return (
      <ContainerProvider value={system}>
        <StoreProvider>
          <AgentProvider agent={undefined}>
            <OpenIDCredentialRecordProvider>
              <ThemeProvider value={theme}>
                <AnimatedComponentsProvider value={animatedComponents}>
                  <AuthProvider>
                    <NetworkProvider>
                      <ActivityProvider>
                        <StatusBar
                          hidden={false}
                          barStyle="light-content"
                          backgroundColor={theme.ColorPallet.brand.primary}
                          translucent={false}
                        />
                        <NetInfo />
                        <ErrorModal />
                        <TourProvider tours={tours} overlayColor={'gray'} overlayOpacity={0.7}>
                          <RootStack />
                        </TourProvider>
                        <Toast topOffset={15} config={toastConfig} />
                      </ActivityProvider>
                    </NetworkProvider>
                  </AuthProvider>
                </AnimatedComponentsProvider>
              </ThemeProvider>
            </OpenIDCredentialRecordProvider>
          </AgentProvider>
        </StoreProvider>
      </ContainerProvider>
    )
  }

  return AppComponent
}
export default App
