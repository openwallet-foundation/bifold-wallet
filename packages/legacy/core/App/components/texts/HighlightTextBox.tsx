import React from 'react'
import { StyleSheet, View } from 'react-native'

import { useTheme } from '../../contexts/theme'
import { ThemedText } from './ThemedText'

export interface TextBoxProps {
  children: string
}

const offset = 10

const HighlightTextBox: React.FC<TextBoxProps> = ({ children }) => {
  const { ColorPallet, OnboardingTheme } = useTheme()
  const style = StyleSheet.create({
    icon: {
      marginRight: offset,
    },
    container: {
      flexDirection: 'row',
      backgroundColor: ColorPallet.brand.primaryBackground,
    },
    accentBox: {
      marginRight: offset,
      backgroundColor: ColorPallet.brand.highlight,
      width: 8,
    },
    headerText: {
      ...OnboardingTheme.bodyText,
      flexShrink: 1,
    },
  })
  return (
    <View style={style.container}>
      <View style={style.accentBox} />
      <ThemedText style={[style.headerText, { paddingTop: offset, paddingBottom: offset }]}>{children}</ThemedText>
    </View>
  )
}

export default HighlightTextBox
