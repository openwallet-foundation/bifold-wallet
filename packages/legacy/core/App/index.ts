/* eslint-disable import/no-cycle */
import type { OnboardingStyleSheet } from './screens/Onboarding'

import { Agent } from '@credo-ts/core'
import AgentProvider from '@credo-ts/react-hooks'

import App from './App'
import * as components from './components'
import { Button as IButton, ButtonImpl as Button, ButtonType } from './components/buttons/Button'
import IconButton, { ButtonLocation } from './components/buttons/IconButton'
import BulletPoint from './components/inputs/BulletPoint'
import CheckBoxRow from './components/inputs/CheckBoxRow'
import ContentGradient from './components/misc/ContentGradient'
import CredentialCard from './components/misc/CredentialCard'
import InfoBox, { InfoBoxType } from './components/misc/InfoBox'
import ErrorModal from './components/modals/ErrorModal'
import NetInfo from './components/network/NetInfo'
import Record from './components/record/Record'
import InfoTextBox from './components/texts/InfoTextBox'
import Link from './components/texts/Link'
import Text from './components/texts/Text'
import { ToastType } from './components/toast/BaseToast'
import toastConfig from './components/toast/ToastConfig'
import { AttachTourStep } from './components/tour/AttachTourStep'
import { credentialOfferTourSteps } from './components/tour/CredentialOfferTourSteps'
import { credentialsTourSteps } from './components/tour/CredentialsTourSteps'
import { homeTourSteps } from './components/tour/HomeTourSteps'
import { proofRequestTourSteps } from './components/tour/ProofRequestTourSteps'
import { TourBox } from './components/tour/TourBox'
import LimitedTextInput from './components/inputs/LimitedTextInput'
import HomeFooterView from './components/views/HomeFooterView'
import KeyboardView from './components/views/KeyboardView'
import NotificationListItem from './components/listItems/NotificationListItem'
import * as contexts from './contexts'
import { AutoLockTime, ActivityProvider, useActivity } from './contexts/activity'
import { AuthProvider } from './contexts/auth'
import { NetworkProvider } from './contexts/network'
import { useTour } from './contexts/tour/tour-context'
import { TourProvider } from './contexts/tour/tour-provider'
import AttemptLockout from './screens/AttemptLockout'
import Developer from './screens/Developer'
import OnboardingPages from './screens/OnboardingPages'
import Preface from './screens/Preface'
import Splash from './screens/Splash'
import Terms from './screens/Terms'
import UseBiometry from './screens/UseBiometry'
import { loadLoginAttempt } from './services/keychain'
import { BifoldLogger } from './services/logger'
import * as types from './types'
import Scan from './screens/Scan'
import Onboarding from './screens/Onboarding'
import { DefaultScreenOptionsDictionary, useDefaultStackOptions } from './navigators/defaultStackOptions'
import { PINRules, walletTimeout } from './constants'
import { CredentialListFooterProps } from './types/credential-list-footer'
import { OpenIDCredentialRecordProvider } from './modules/openid/context/OpenIDCredentialRecordProvider'
import { defaultConfig, defaultHistoryEventsLogger } from './container-impl'
import useInitializeAgent from './hooks/initialize-agent'
import { DefaultScreenLayoutOptions } from './navigators/defaultLayoutOptions'

export * from './navigators'
export * from './services/storage'
export * from './types/attestation'
export { LocalStorageKeys } from './constants'
export * from './services/storage'
export { initLanguages, initStoredLanguage, translationResources, Locales } from './localization'
export { defaultState, mergeReducers, StoreProvider, StoreContext, useStore } from './contexts/store'
export { default as Store, DispatchAction, reducer } from './contexts/reducers/store'
export { useDeepLinks } from './hooks/deep-links'
export { Assets as ImageAssets } from './theme'
export { ThemeProvider, useTheme } from './contexts/theme'
export { AnimatedComponentsProvider, useAnimatedComponents } from './contexts/animated-components'
export { ColorPallet } from './theme'
export { animatedComponents } from './animated-components'
export { theme } from './theme'
export { useAuth } from './contexts/auth'
export { useNetwork } from './contexts/network'
export { NavigationTheme } from './theme'
export { testIdWithKey, testIdForAccessabilityLabel } from './utils/testable'
export { Screens, Stacks, TabStacks } from './types/navigators'
export { createStyles } from './screens/OnboardingPages'
export { statusBarStyleForColor, StatusBarStyles } from './utils/luminance'
export { BifoldError } from './types/error'
export { EventTypes } from './constants'
export { migrateToAskar } from './utils/migration'
export { createLinkSecretIfRequired, getAgentModules } from './utils/agent'
export { removeExistingInvitationIfRequired, connectFromScanOrDeepLink, formatTime, useCredentialConnectionLabel, getConnectionName } from './utils/helpers'
export { isValidAnonCredsCredential, getCredentialIdentifiers } from './utils/credential'
export { buildFieldsFromAnonCredsCredential } from './utils/oca'

export type { AnimatedComponents } from './animated-components'
export type {
  ISVGAssets,
  IFontAttributes,
  IInputAttributes,
  IInputs,
  ITextTheme,
  IBrandColors,
  ISemanticColors,
  INotificationColors,
  IGrayscaleColors,
  IColorPallet,
  IAssets,
  ITheme,
} from './theme'
export type { PersistentState } from './types/state'
export type { BifoldAgent } from './utils/agent'
export type { TourStep, RenderProps } from './contexts/tour/tour-context'
export type { GenericFn } from './types/fn'
export type {
  AuthenticateStackParams,
  OnboardingStackParams,
  NotificationStackParams,
  ContactStackParams,
} from './types/navigators'
export type { OnboardingStyleSheet }
export type { WalletSecret } from './types/security'
export type { ReducerAction } from './contexts/reducers/store'
export type {
  State,
  Onboarding as OnboardingState,
  LoginAttempt as LoginAttemptState,
  Preferences as PreferencesState,
  Migration as MigrationState,
  Tours as ToursState,
} from './types/state'

export type { InlineMessageProps } from './components/inputs/InlineErrorText'

export { InlineErrorPosition } from './types/error'

export type { CredentialListFooterProps }
export * from './container-api'
export { MainContainer } from './container-impl'

export type { ScreenLayoutConfig } from './types/navigators'
export type { HistoryEventsLoggerConfig } from './types/config'

export {
  App,
  Agent,
  AgentProvider,
  AuthProvider,
  NetworkProvider,
  TourProvider,
  useTour,
  AttachTourStep,
  TourBox,
  defaultConfig,
  defaultHistoryEventsLogger,
  homeTourSteps,
  credentialsTourSteps,
  credentialOfferTourSteps,
  proofRequestTourSteps,
  ButtonType,
  ButtonLocation,
  CheckBoxRow,
  CredentialCard,
  ContentGradient,
  ErrorModal,
  IconButton,
  ActivityProvider,
  useActivity,
  InfoTextBox,
  InfoBox,
  InfoBoxType,
  Link,
  AutoLockTime,
  ToastType,
  toastConfig,
  NetInfo,
  OnboardingPages,
  OpenIDCredentialRecordProvider,
  NotificationListItem,
  useDefaultStackOptions,
  useInitializeAgent,
  Splash,
  Developer,
  Terms,
  Preface,
  HomeFooterView as HomeContentView,
  UseBiometry,
  AttemptLockout,
  Record,
  Scan,
  Onboarding,
  types,
  components,
  contexts,
  Text,
  loadLoginAttempt,
  Button,
  BifoldLogger,
  LimitedTextInput,
  KeyboardView,
  BulletPoint,
  PINRules,
  walletTimeout,
  DefaultScreenOptionsDictionary,
  DefaultScreenLayoutOptions,
}
export type { IButton }
