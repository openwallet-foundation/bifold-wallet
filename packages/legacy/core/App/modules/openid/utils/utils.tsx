import { getDomainFromUrl } from '@credo-ts/core'
import { Attribute, Field } from '@hyperledger/aries-oca/build/legacy'
import { W3cCredentialDisplay } from '../types'

/**
 * Converts a camelCase string to a sentence format (first letter capitalized, rest in lower case).
 * i.e. sanitizeString("helloWorld")  // returns: 'Hello world'
 */
export function sanitizeString(str: string) {
  const result = str.replace(/([a-z0-9])([A-Z])/g, '$1 $2').replaceAll('_', ' ')
  let words = result.split(' ')
  words = words.map((word, index) => {
    if (index === 0 || word.toUpperCase() === word) {
      return word.charAt(0).toUpperCase() + word.slice(1)
    }
    return word.charAt(0).toLowerCase() + word.slice(1)
  })
  return words.join(' ')
}

export function getHostNameFromUrl(url: string) {
  try {
    return getDomainFromUrl(url)
  } catch (error) {
    throw new Error(`Error getting hostname from url: ${error}`)
  }
}

export const buildFieldsFromOpenIDTemplate = (data: { [key: string]: unknown }): Array<Field> => {
  const fields = []
  for (const key of Object.keys(data)) {
    // omit id and type
    if (key === 'id' || key === 'type') continue

    let pushedVal: string | number | null = null
    if (typeof data[key] === 'string' || typeof data[key] === 'number') {
      pushedVal = data[key] as string | number | null
    }
    fields.push(new Attribute({ name: key, value: pushedVal }))
  }
  return fields
}

export function formatDate(input: string | Date): string {
  const date = input instanceof Date ? input : new Date(input)
  return date.toLocaleString('en-US', {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
    hour: '2-digit',
    minute: '2-digit',
    hour12: true,
  })
}

export const getAttributeField = (display: W3cCredentialDisplay, searchKey: string): Attribute | undefined => {
  for (const [key, value] of Object.entries(display.attributes)) {
    if (searchKey === key) {
      return new Attribute({
        name: key,
        value: value as string | number | null,
        mimeType: typeof value === 'number' ? 'text/number' : 'text/plain',
      })
    }
  }
  return undefined
}
