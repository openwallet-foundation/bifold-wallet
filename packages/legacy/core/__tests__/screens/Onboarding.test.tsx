import { fireEvent, render } from '@testing-library/react-native'
import React from 'react'

import * as themeContext from '../../App/contexts/theme' // note we're importing with a * to import all the exports
import Onboarding, { OnboardingStyleSheet } from '../../App/screens/Onboarding'
import { createCarouselStyle, createPageWith } from '../../App/screens/OnboardingPages'
import { OnboardingTheme, theme } from '../../App/theme'
import CredentialList from '../../App/assets/img/credential-list.svg'
import { testIdWithKey } from '../../App/utils/testable'
import { ThemedText } from '../../App/components/texts/ThemedText'

export const carousel: OnboardingStyleSheet = createCarouselStyle(OnboardingTheme)

const pages = [
  <>
    <ThemedText testID={'bodyText'}>Hello</ThemedText>
  </>,
  <>
    <ThemedText testID={'bodyText'}>World</ThemedText>
  </>,
]

describe('Onboarding Screen', () => {
  test('Renders correctly', () => {
    jest.spyOn(themeContext, 'useTheme').mockImplementation(() => theme)
    const tree = render(<Onboarding pages={pages} nextButtonText="Next" previousButtonText="Back" style={carousel} />)

    expect(tree).toMatchSnapshot()
  })

  test('Pages exist', async () => {
    jest.spyOn(themeContext, 'useTheme').mockImplementation(() => theme)
    const { findAllByTestId } = render(
      <Onboarding pages={pages} nextButtonText="Next" previousButtonText="Back" style={carousel} />
    )
    const foundPages = await findAllByTestId('bodyText')

    expect(foundPages).toHaveLength(2)
  })

  test('Onboarding Developer mode', async () => {
    jest.spyOn(themeContext, 'useTheme').mockImplementation(() => theme)
    const testFunc = jest.fn()
    const tree = render(
      <Onboarding
        pages={[createPageWith(CredentialList, 'test', 'body', {}, true, testFunc)]}
        nextButtonText="Next"
        previousButtonText="Back"
        style={carousel}
      />
    )
    expect(tree).toMatchSnapshot()

    const devModeButton = await tree.getByTestId(testIdWithKey('DeveloperModeTouch'))
    fireEvent(devModeButton, 'press')
    expect(testFunc).toBeCalledTimes(1)
  })
})
