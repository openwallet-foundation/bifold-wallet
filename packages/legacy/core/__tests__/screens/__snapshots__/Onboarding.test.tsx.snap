// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Onboarding Screen Onboarding Developer mode 1`] = `
<RNCSafeAreaView
  edges={
    Object {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "off",
    }
  }
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000000",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        <ScrollView
          style={
            Object {
              "padding": 20,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
              }
            }
          >
            <
              style={
                Object {
                  "height": 180,
                  "width": 180,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "marginBottom": 20,
              }
            }
          >
            <TouchableWithoutFeedback
              onPress={[MockFunction]}
              testID="com.ariesbifold:id/DeveloperModeTouch"
            >
              <ThemedText
                style={Object {}}
                testID="com.ariesbifold:id/HeaderText"
              >
                test
              </ThemedText>
            </TouchableWithoutFeedback>
            <ThemedText
              style={
                Array [
                  Object {
                    "flexShrink": 1,
                  },
                  Object {
                    "marginTop": 25,
                  },
                ]
              }
              testID="com.ariesbifold:id/BodyText"
            >
              body
            </ThemedText>
          </View>
        </ScrollView>,
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onViewableItemsChanged={[Function]}
    pagingEnabled={true}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={16}
    showsHorizontalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "width": 750,
      }
    }
    viewabilityConfig={
      Object {
        "viewAreaCoveragePercentThreshold": 60,
      }
    }
    viewabilityConfigCallbackPairs={
      Array [
        Object {
          "onViewableItemsChanged": [Function],
          "viewabilityConfig": Object {
            "viewAreaCoveragePercentThreshold": 60,
          },
        },
      ]
    }
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "width": 750,
              },
              Object {
                "backgroundColor": "#000000",
                "flexDirection": "column",
              },
            ]
          }
        >
          <RCTScrollView
            style={
              Object {
                "padding": 20,
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                  }
                }
              >
                <
                  style={
                    Object {
                      "height": 180,
                      "width": 180,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "marginBottom": 20,
                  }
                }
              >
                <Text
                  accessibilityState={
                    Object {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessible={true}
                  focusable={true}
                  maxFontSizeMultiplier={2}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 18,
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  testID="com.ariesbifold:id/DeveloperModeTouch"
                >
                  test
                </Text>
                <Text
                  maxFontSizeMultiplier={2}
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 18,
                        "fontWeight": "normal",
                      },
                      Array [
                        Object {
                          "flexShrink": 1,
                        },
                        Object {
                          "marginTop": 25,
                        },
                      ],
                    ]
                  }
                  testID="com.ariesbifold:id/BodyText"
                >
                  body
                </Text>
              </View>
            </View>
          </RCTScrollView>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexShrink": 1,
        "marginBottom": 30,
      }
    }
  >
    <View
      accessibilityElementsHidden={true}
      accessibilityLabel="Global.Back"
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        Object {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 44,
          "left": 44,
          "right": 44,
          "top": 44,
        }
      }
      importantForAccessibility="no-hide-descendants"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="com.ariesbifold:id/Back"
    >
      <Text
        maxFontSizeMultiplier={2}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "fontWeight": "normal",
            },
            Array [
              Object {
                "color": "#42803E",
                "fontSize": 18,
                "fontWeight": "bold",
              },
              Object {
                "color": "transparent",
                "paddingRight": 20,
              },
            ],
          ]
        }
      >
        Back
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "bottom": 20,
            "flexDirection": "row",
            "position": "absolute",
          },
          Object {
            "position": "relative",
            "top": 0,
          },
        ]
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(66, 128, 62, 1)",
            "borderColor": "#42803E",
            "borderRadius": 5,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 10,
            "marginHorizontal": 5,
            "opacity": 1,
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": 10,
          }
        }
      />
    </View>
    <View
      accessibilityElementsHidden={true}
      accessibilityLabel="Global.Next"
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        Object {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 44,
          "left": 44,
          "right": 44,
          "top": 44,
        }
      }
      importantForAccessibility="no-hide-descendants"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="com.ariesbifold:id/Next"
    >
      <Text
        maxFontSizeMultiplier={2}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "fontWeight": "normal",
            },
            Array [
              Object {
                "color": "#42803E",
                "fontSize": 18,
                "fontWeight": "bold",
              },
              Object {
                "color": "transparent",
                "paddingLeft": 20,
              },
            ],
          ]
        }
      >
        Next
      </Text>
    </View>
  </View>
</RNCSafeAreaView>
`;

exports[`Onboarding Screen Renders correctly 1`] = `
<RNCSafeAreaView
  edges={
    Object {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "off",
    }
  }
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000000",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={
      Array [
        <React.Fragment>
          <ThemedText
            testID="bodyText"
          >
            Hello
          </ThemedText>
        </React.Fragment>,
        <React.Fragment>
          <ThemedText
            testID="bodyText"
          >
            World
          </ThemedText>
        </React.Fragment>,
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onViewableItemsChanged={[Function]}
    pagingEnabled={true}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={16}
    showsHorizontalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "width": 750,
      }
    }
    viewabilityConfig={
      Object {
        "viewAreaCoveragePercentThreshold": 60,
      }
    }
    viewabilityConfigCallbackPairs={
      Array [
        Object {
          "onViewableItemsChanged": [Function],
          "viewabilityConfig": Object {
            "viewAreaCoveragePercentThreshold": 60,
          },
        },
      ]
    }
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "width": 750,
              },
              Object {
                "backgroundColor": "#000000",
                "flexDirection": "column",
              },
            ]
          }
        >
          <Text
            maxFontSizeMultiplier={2}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 18,
                  "fontWeight": "normal",
                },
                undefined,
              ]
            }
            testID="bodyText"
          >
            Hello
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "width": 750,
              },
              Object {
                "backgroundColor": "#000000",
                "flexDirection": "column",
              },
            ]
          }
        >
          <Text
            maxFontSizeMultiplier={2}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 18,
                  "fontWeight": "normal",
                },
                undefined,
              ]
            }
            testID="bodyText"
          >
            World
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexShrink": 1,
        "marginBottom": 30,
      }
    }
  >
    <View
      accessibilityElementsHidden={true}
      accessibilityLabel="Global.Back"
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        Object {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 44,
          "left": 44,
          "right": 44,
          "top": 44,
        }
      }
      importantForAccessibility="no-hide-descendants"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="com.ariesbifold:id/Back"
    >
      <Text
        maxFontSizeMultiplier={2}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "fontWeight": "normal",
            },
            Array [
              Object {
                "color": "#42803E",
                "fontSize": 18,
                "fontWeight": "bold",
              },
              Object {
                "color": "transparent",
                "paddingRight": 20,
              },
            ],
          ]
        }
      >
        Back
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "bottom": 20,
            "flexDirection": "row",
            "position": "absolute",
          },
          Object {
            "position": "relative",
            "top": 0,
          },
        ]
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(66, 128, 62, 1)",
            "borderColor": "#42803E",
            "borderRadius": 5,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 10,
            "marginHorizontal": 5,
            "opacity": 1,
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": 10,
          }
        }
      />
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgba(255, 255, 255, 1)",
            "borderColor": "#42803E",
            "borderRadius": 5,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 10,
            "marginHorizontal": 5,
            "opacity": 1,
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": 10,
          }
        }
      />
    </View>
    <View
      accessibilityElementsHidden={false}
      accessibilityLabel="Global.Next"
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        Object {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 44,
          "left": 44,
          "right": 44,
          "top": 44,
        }
      }
      importantForAccessibility="auto"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="com.ariesbifold:id/Next"
    >
      <Text
        maxFontSizeMultiplier={2}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "fontWeight": "normal",
            },
            Array [
              Object {
                "color": "#42803E",
                "fontSize": 18,
                "fontWeight": "bold",
              },
              Object {
                "color": "#42803E",
                "paddingLeft": 20,
              },
            ],
          ]
        }
      >
        Next
      </Text>
    </View>
  </View>
</RNCSafeAreaView>
`;
